### DISCUSSION TITLE
ts_tree_edit: can I use only bytes and no TSPosition?

### DISCUSSION THREAD

**Discussion by ccw630** [tree-sitter/tree-sitter#2955] [Q&A]
mirror question of https://github.com/tree-sitter/tree-sitter/issues/210

> In applications like text editors, you often need to re-parse a file after its source code has changed. Tree-sitter is designed to support this use case efficiently. There are two steps required. First, you must edit the syntax tree, which adjusts the ranges of its nodes so that they stay in sync with the code.
```
typedef struct {
  uint32_t start_byte;
  uint32_t old_end_byte;
  uint32_t new_end_byte;
  TSPoint start_point;
  TSPoint old_end_point;
  TSPoint new_end_point;
} TSInputEdit;

void ts_node_edit(TSNode *, const TSInputEdit *);
```

I've tried pass three correct bytes and three `dummyPoint`s(`[0:0]`) to the edit, and the reparse result seems all good: the tree, the position of new nodes..

**Comment by jmehnle** [comment]
I, too, am interested in this functionality. @maxbrunsfeld, how optional are the `…_point` arguments, and what are the implications of passing nonsensical values for them?
